File: booking-service/src/main/java/com/restaurant/booking/booking/service/service/RatingService.java
package com.restaurant.booking.booking.service.service;

import com.restaurant.booking.booking.model.Rating;
import com.restaurant.booking.booking.model.RatingCreationRequest;
import com.restaurant.booking.booking.model.RatingStatus;

import java.util.List;

public interface RatingService {

    Rating save(Rating rating);

    Rating createRating(RatingCreationRequest requestCreation);

    List<Rating> getRestaurantRatings(String restaurantId);

    List<Rating> getFlagedRatings();

    Rating changeRatingStatus(RatingStatus ratingStatus, String ratingId);
}

File: front-end/src/services/RatingService.js
import {authAxios} from '@/http-common'

class RatingService {

    createRating(data){
        return authAxios.post('/ratings', data, {
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token')
            }
        })
    }

    getRestaurantRatings(restaurantId){        
        return authAxios.get('/ratings/restaurant/' + restaurantId, {
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token')
            }
        })
    }

    getFlaggedRatings(){        
        return authAxios.get('/ratings/flag', {
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token')
            }
        })
    }

    flagRating(ratingId){        
        return authAxios.put('/ratings/' + ratingId, {
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token')
            }
        })
    }

    changeRatingStatus(ratingId, ratingStatus){        
        return authAxios.put('/ratings/' + ratingId + '/status/' + ratingStatus, {
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token')
            }
        })
    }
}

export default new RatingService()